dom/fuzzer/modules/web-speech.js

```diff
@@ -0,0 +1,173 @@
+            utterances.push(myutterance);
+          }],
+      [1, function(){
+            // set attribute on an utterance
+            if (!utterances.length) return;
+            cmd.push(JS.setAttribute(Random.pick(utterances), utteranceAttributes) + ';');
+          }],
+      [1, function(){
+            // call something on SpeechSynthesis
+            cmd.push(JS.methodCall('window.speechSynthesis', synthMethods) + ';');
+          }],
+      [1, function(){
+            // set an utterance event
+            if (!utterances.length) return;
+            cmd.push(Random.pick(utterances) + ".addEventListener" + JS.methodHead([simpleSource(Random.pick(utteranceEvents)), "function(e) { " + fuzzSubCommand() + "}"]) + ';');
+          }],
+/*
+      [1, function(){
+            // create a recognition
+            var myrecognition = Things.reserve();
+            cmd.push(myrecognition + ' = new SpeechRecognition();');
+            recognitions.push(myrecognition);
+          }],
+      [1, function(){
+            // set attribute on a recognition
+            if (!recognitions.length) return;
+            cmd.push(JS.setAttribute(Random.pick(recognitions), recognitionAttributes) + ';');
+          }],
+      [1, function(){
+            // call something on SpeechRecognition
+            if (!recognitions.length) return;
+            cmd.push(JS.methodCall(Random.pick(recognitions), recognitionMethods) + ';');
+          }],
+      [1, function(){
+            // set a recognition event
+            if (!recognitions.length) return;
+            cmd.push(Random.pick(recognitions) + ".addEventListener" + JS.methodHead([simpleSource(Random.pick(recognitionEvents)), "function(e) { " + fuzzSubCommand() + "}"]) + ';');
+          }],
+      [1, function(){
+            // create a grammar
+            var mygrammar = Things.reserve();
+            cmd.push(mygrammar + ' = new SpeechGrammar();');
+            grammars.push(mygrammar);
+          }],
+      [1, function(){
+            // set attribute on a grammar
+            var g = getGrammar();
+            if (g === undefined) return;
+            cmd.push(JS.setAttribute(g, grammarAttributes) + ';');
+          }],
+      [1, function(){
+            // create a grammar list
+            var mygrammarl = Things.reserve();
+            cmd.push(mygrammarl + ' = new SpeechGrammarList();');
+            grammarLists.push(mygrammarl);
+          }],
+      [1, function(){
+            // call something on SpeechGrammarList
+            if (!grammarLists.length) return;
+            cmd.push(JS.methodCall(Random.pick(grammarLists), grammarListMethods) + ';');
+          }],
+*/
+    ]);
+    return cmd;
+  };
+
+  var getGrammar = function(){
+    var poss = [];
+    if (grammarLists.length)
+      poss.push([10, function(){ var gl = Random.pick(grammarLists); return gl + '[' + rnd(1024) + '%' + gl + '.length]'; }]);
```

Comment:
Is this 1024%length thing going to be stable through testcase reduction? Not sure if there's a way to make it more stable given the API.